sklearn/utils/_encode.py

```diff
@@ -220,7 +220,13 @@ def _encode(values, *, uniques, check_unknown=True):
     encoded : ndarray
         Encoded values
     """
-    if values.dtype.kind in "OUS":
+    xp, is_array_api_compliant = get_namespace(values, uniques)
+    if is_array_api_compliant:
+        dtype_kind = _convert_to_numpy(values, xp).dtype.kind
```

Comment:
This is a bit expensive just to inspect a dtype kind.

We should instead change the `if dtype_kind in "OUS"` condition to rely on a combo or `is_array_api_compliant` and `xp.isdtype` with the `kind` parameter, see:

- https://data-apis.org/array-api/latest/API_specification/generated/array_api.isdtype.html#isdtype